latch_cli/services/cp/utils.py

```diff
@@ -0,0 +1,199 @@
+import os
+import re
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+from typing import Dict, TypedDict
+from urllib.parse import urlparse
+
+import gql
+
+from latch_cli.config.user import user_config
+from latch_cli.services.cp.exceptions import AuthenticationError, PathResolutionError
+
+
+def get_auth_header() -> Dict[str, str]:
+    sdk_token = user_config.token
+    execution_token = os.environ.get("FLYTE_INTERNAL_EXECUTION_ID")
+
+    if sdk_token is not None and sdk_token != "":
+        headers = {"Authorization": f"Latch-SDK-Token {sdk_token}"}
+    elif execution_token is not None:
+        headers = {"Authorization": f"Latch-Execution-Token {execution_token}"}
+    else:
+        raise AuthenticationError("Unable to find authentication credentials.")
+
+    return headers
+
+
+is_valid_path_expr = re.compile(r"^(latch)?://")
+
+
+def is_remote_path(path: str) -> bool:
+    return is_valid_path_expr.match(path) is not None
+
+
+legacy_expr = re.compile(r"^(account_root|mount)/([^/]+)(?:/+(.*))?$")
+scheme_expr = re.compile(
+    r"^(?:"
+    r"(?P<full>latch://[^/]*(/+.*)?)"
+    r"|(?P<missing_latch>://[^/]*(/+.*)?)"
+    r"|(?P<missing_scheme_with_leading_slash>/+.*)"
+    r"|(?P<missing_scheme_without_leading_slash>[^/]+.*)"
+    r")$"
+)
+domain_expr = re.compile(
```

Comment:
```suggestion
domain = re.compile(
    r"""
    ^(
	    (?P<empty>/) |
	    (   
	        ((?P<account>\d+)\.account) |
	        (?P<shared>shared) |
	        (shared\.(?P<shared_account>\d+)\.account) |
	        ((?P<mount>[^/]+)\.mount) |
	        ((?P<node_id>\d+)\.node)
	    )/
	   )
    """,
    re.VERBOSE,
)
```

+ do not support `archive` anymore, instead use `node_id.node`

Reply:
also I'd drop all the capture groups since we don't actually use them?
```
domain = re.compile(
    r"""
    ^(
	    (   
					| # empty
	        (\d+\.account) |
	        (shared) |
	        (shared\.\d+\.account) |
	        ([^/]+\.mount) |
	        (\d+\.node)
	    )/
	   )
    """,
    re.VERBOSE,
)
```